flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/row/ParquetRowDataWriter.java

```diff
@@ -381,9 +381,16 @@ private MapWriter(LogicalType keyType, LogicalType valueType, GroupType groupTyp
 
         @Override
         public void write(RowData row, int ordinal) {
-            recordConsumer.startGroup();
+            writeMapData(row.getMap(ordinal));
+        }
 
-            MapData mapData = row.getMap(ordinal);
+        @Override
+        public void write(ArrayData arrayData, int ordinal) {
```

Comment:
NIT: Not sure why those write(ArrayData arrayData, int ordinal) methods didn't get implemented. It should be trivial to do it while building those Writers. Do you have any hints about the background info?